src/Vgrep/Widget/Results.hs

```diff
@@ -69,8 +75,10 @@ pageDown = do
     modifying files (repeatedly moveDown)
 
 repeatedly :: (a -> Maybe a) -> a -> a
-repeatedly f a = case f a of Just a' -> repeatedly f a'
-                             Nothing -> a
```

Comment:
It's probably unnecessary here. Sometimes this sort of wrapper transformation is used to share a parameter. Whether this is truly useful depends on how the function will mostly be used.

This is because GHC will only inline fully applied functions, so in your case you will get no inlining for e.g. `repeatedly f . doStuff`, while you might get inlining for `repeatedly f x`. This affects how further rewrite rules apply, and inlining definitions early might open up more possibilities. The actual definition of `foldr` uses this technique for example [(source link)](http://hackage.haskell.org/package/base-4.8.2.0/docs/src/GHC.Base.html#foldr).